### PR TITLE
🔧 (version): FirmwareKit use os version generated from cmake

### PR DESCRIPTION
**Pull request by YannLocatelli**
Need: 

- [x] #1023
- [ ] Add workflow to check version are correspondant in config and generated binary
- [x] Rename `Firmware_Version` by `Version`
- [x] ~~Fix use of `settings.json`~~
- [ ] Refactor getCurrentVersionFromFile

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,6 +65,13 @@ if(ENABLE_LOG_DEBUG AND ENABLE_SYSTEM_STATS)
 	add_definitions(-DMBED_HEAP_STATS_ENABLED)
 endif(ENABLE_LOG_DEBUG AND ENABLE_SYSTEM_STATS)
 
+# Firmware version
+option(FIRMWARE_VERSION "Set firmware version" "0.0.0")
+
+if(FIRMWARE_VERSION)
```
Comment (ladislas) [inline CMakeLists.txt:71]: as you give the initial value `0.0.0` to `FIRMWARE_VERSION`, the condition is always true.
the `if` can be removed

File: libs/FirmwareKit/source/FirmwareKit.cpp [hunk 2]
```diff
@@ -10,7 +10,12 @@ using namespace leka;
 
 auto FirmwareKit::getCurrentVersion() -> FirmwareVersion
 {
+#if defined(FIRMWARE_VERSION)
```
Comment (ladislas) [inline libs/FirmwareKit/source/FirmwareKit.cpp:13]: I'm not a big fan of macros, especially this deep in the implementation. 

my understanding is that `#if defined(FIRMWARE_VERSION)` is always true, even if the value is the empty string `""`

If I'm not mistaken, then the other path is never reached and can be removed.

what is the rational for keeping the `getCurrentVersionFromFile` part?

I'm really careful with these macro magic tricks. My take would be "force the firmware version everywhere, there is no reason to not define/use it and skip all the checks".

**Review by ladislas — CHANGES_REQUESTED** [review]
Looks nice but I'm worried about the use of macros like that. To be discussed tomorrow :)

File: CMakeLists.txt (hunk 1 above)
Comment (YannLocatelli) [reply CMakeLists.txt:71]: ok

File: libs/FirmwareKit/source/FirmwareKit.cpp (hunk 2 above)
Comment (YannLocatelli) [reply libs/FirmwareKit/source/FirmwareKit.cpp:13]: > my understanding is that #if defined(FIRMWARE_VERSION) is always true, even if the value is the empty string ""

Not it is not, if it is the case I will not implement it this way.

But I get the point of the reliability of macro, so let me know what is your suggestion

**Comment by YannLocatelli** [comment]
- [ ] Add workflow to check version are correspondant in config and generated binary
- [ ] Rename `Firmware_Version` by `Version`

**Comment by YannLocatelli** [comment]
Replaced by #1069

**Pull request closed without merging**